src/rosdep2/platforms/pip.py

```diff
@@ -70,6 +84,45 @@ def get_pip_command():
     return None
 
 
+def externally_managed_installable():
+    """
+    PEP 668 enacted in Python 3.11 blocks pip from working in "externally
+    managed" environments such as operating systems with included package
+    managers. If we're on Python 3.11 or greater, we need to check that pip
+    is configured to allow installing system-wide packages with the
+    flagrantly named "break system packages" config option or environment
+    variable.
+    """
+
+    # This doesn't affect Python versions before 3.11
+    if sys.version_info < (3, 11):
+        return True
+
+    if (
+            'PIP_BREAK_SYSTEM_PACKAGES' in os.environ and
+            os.environ['PIP_BREAK_SYSTEM_PACKAGES'].lower() in ('yes', '1', 'true')
+    ):
+        return True
+
+    # Check the same configuration directories as pip does per
+    # https://pip.pypa.io/en/stable/topics/configuration/
+    pip_config = ConfigParser()
+    if 'XDG_CONFIG_DIRS' in os.environ:
+        for xdg_dir in os.environ['XDG_CONFIG_DIRS'].split(':'):
+            pip_config_file = Path(xdg_dir) / 'pip' / 'pip.conf'
+            pip_config.read(pip_config_file)
+            if pip_config['install']['break-system-packages']:
+                return True
+
+    fallback_config = Path('/etc/pip.conf')
+    pip_config.read(fallback_config)
+    if pip_config['install']['break-system-packages']:
```

Comment:
I gave it a try and it seems like, if:

1. the config file doesn't exist; or
2. the `[install]` section doesn't exist; or
3. a `break-system-package` value isn't specified.

then `pip_config[...]` raises a `KeyError`. Looks like you could avoid checking all of that by using `getboolean()`:

```py
pip_config.getboolean('install', 'break-system-packages', fallback=False)
```

Reply:
Good catch!

Reply:
Resolved in [8e94ef1](https://github.com/ros-infrastructure/rosdep/pull/979/commits/8e94ef1ffb01394b23cea9dcc12fbb6fc92f10d2)